test/e2e/grpc/client.test.ts

```diff
@@ -318,4 +319,47 @@ describe('grpc/client', () => {
       await stream2.stop();
     });
   });
-});
+
+  describe('distributed lock', () => {
+    it('should be able to acquire a new lock and unlock', async () => {
+      const tryLock = await client.lock.tryLock("redislock", "resourceId", "owner1", 1000);
+      expect(tryLock.success).toEqual(true);
+      const unlock = await client.lock.unlock("redislock", "resourceId", "owner1");
+      expect(unlock.status).toEqual(LockStatus.Success);
+    });
+
+    it('should be not be able to unlock when the lock is no acquired', async () => {
+      const unlock = await client.lock.unlock("redislock", "resourceId", "owner1");
+      expect(unlock.status).toEqual(LockStatus.LockUnexist);
+    });
+
+    it('should be able to acquire a lock after the previous lock is expired', async () => {
+      let tryLock = await client.lock.tryLock("redislock", "resourceId", "owner1", 5);
+      expect(tryLock.success).toEqual(true);
+      await new Promise(resolve => setTimeout(resolve, 2000));
+      tryLock = await client.lock.tryLock("redislock", "resourceId", "owner2", 5);
+      expect(tryLock.success).toEqual(false);
+    });
+
+    it('should not be able to acquire a lock when the same lock is acquired by the other owner', async () => {
+      const tryLockOne = await client.lock.tryLock("redislock", "resourceId1", "owner1", 5);
+      expect(tryLockOne.success).toEqual(true);
+      const tryLockTwo = await client.lock.tryLock("redislock", "resourceId1", "owner2", 5);
+      expect(tryLockTwo.success).toEqual(false);
+    });
+
+    it('should be able to acquire a lock when a different lock is acquired by the other owner', async () => {
+      const tryLockOne = await client.lock.tryLock("redislock", "resourceId2", "owner1", 5);
+      expect(tryLockOne.success).toEqual(true);
+      const tryLockTwo = await client.lock.tryLock("redislock", "resourceId3", "owner2", 5);
+      expect(tryLockTwo.success).toEqual(true);
+    });
+
+    it('should not be able to acquire a lock when that lock is acquired by the other owner/process', async () => {
```

Comment:
I think we are missing the test case "should not be able to unlock a lock when that lock is acquired by the other owner/process".